command/agent/agent_endpoint.go

```diff
@@ -333,6 +334,88 @@ func (s *HTTPServer) AgentForceLeaveRequest(resp http.ResponseWriter, req *http.
 	return nil, err
 }
 
+func (s *HTTPServer) AgentPprofRequest(resp http.ResponseWriter, req *http.Request) ([]byte, error) {
+	path := strings.TrimPrefix(req.URL.Path, "/v1/agent/pprof/")
+	switch path {
+	case "":
+		// no root index route
+		return nil, CodedError(404, ErrInvalidMethod)
+	case "cmdline":
+		return s.agentPprof(pprof.CmdReq, resp, req)
+	case "profile":
+		return s.agentPprof(pprof.CPUReq, resp, req)
+	case "trace":
+		return s.agentPprof(pprof.TraceReq, resp, req)
+	default:
+		// Add profile to request
+		values := req.URL.Query()
+		values.Add("profile", path)
+		req.URL.RawQuery = values.Encode()
+
+		// generic pprof profile request
+		return s.agentPprof(pprof.LookupReq, resp, req)
+	}
+}
+
+func (s *HTTPServer) agentPprof(reqType pprof.ReqType, resp http.ResponseWriter, req *http.Request) ([]byte, error) {
+
+	// Parse query param int values
+	seconds, _ := strconv.Atoi(req.URL.Query().Get("seconds"))
+	debug, _ := strconv.Atoi(req.URL.Query().Get("debug"))
+	gc, _ := strconv.Atoi(req.URL.Query().Get("gc"))
```

Comment:
I was mimicking the net/pprof handling of these values which just assign a default value instead of requiring valid params https://golang.org/src/net/http/pprof/pprof.go?s=2316:2368#L145

Reply:
Makes sense - Maybe then I'd add a comment on why the errors are dropped?